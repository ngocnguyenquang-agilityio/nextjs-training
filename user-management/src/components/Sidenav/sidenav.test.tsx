import { fireEvent, render } from '@testing-library/react';

// Components
import SideNav from '.';

const mockUsePathname = jest.fn();

jest.mock('next/navigation', () => ({
  usePathname() {
    return mockUsePathname();
  }
}));

describe('SideNav Component', () => {
  test('Should render successfully', () => {
    const comp = render(<SideNav />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<SideNav />);
    expect(comp).toMatchSnapshot();
  });

  test('Should call usePathname ', async () => {
    const comp = render(<SideNav />);
    const link = comp.queryByTestId('/home') as HTMLAnchorElement;
    await fireEvent.click(link);
    expect(mockUsePathname).toBeCalled();
  });
});
