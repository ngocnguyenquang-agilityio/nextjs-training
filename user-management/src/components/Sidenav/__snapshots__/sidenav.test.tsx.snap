// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-full flex-col px-3 py-4 md:px-2"
      >
        <a
          class="mb-2 flex h-20 items-end justify-start rounded-md bg-blue-600 p-4 md:h-40"
          href="/"
        >
          <div
            class="w-32 text-white md:w-40"
          >
            <div
              class="flex flex-row items-center leading-none text-white"
            >
              <p
                class="text-[24px]"
              >
                User Management
              </p>
            </div>
          </div>
        </a>
        <div
          class="flex grow flex-row justify-between space-x-2 md:flex-col md:space-x-0 md:space-y-2"
        >
          <a
            class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm text-black font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            data-testid="/home"
            href="/home"
          >
            <svg
              aria-hidden="true"
              class="w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Home
            </p>
          </a>
          <a
            class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm text-black font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            data-testid="/users"
            href="/users"
          >
            <svg
              aria-hidden="true"
              class="w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 18.72a9.094 9.094 0 003.741-.479 3 3 0 00-4.682-2.72m.94 3.198l.001.031c0 .225-.012.447-.037.666A11.944 11.944 0 0112 21c-2.17 0-4.207-.576-5.963-1.584A6.062 6.062 0 016 18.719m12 0a5.971 5.971 0 00-.941-3.197m0 0A5.995 5.995 0 0012 12.75a5.995 5.995 0 00-5.058 2.772m0 0a3 3 0 00-4.681 2.72 8.986 8.986 0 003.74.477m.94-3.197a5.971 5.971 0 00-.94 3.197M15 6.75a3 3 0 11-6 0 3 3 0 016 0zm6 3a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0zm-13.5 0a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Users
            </p>
          </a>
          <a
            class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm text-black font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
            data-testid="/techstacks"
            href="/techstacks"
          >
            <svg
              aria-hidden="true"
              class="w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.25 6.75L22.5 12l-5.25 5.25m-10.5 0L1.5 12l5.25-5.25m7.5-3l-4.5 16.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Tech Stacks
            </p>
          </a>
          <div
            class="hidden h-auto w-full grow rounded-md bg-gray-50 md:block"
          />
          <button
            class="flex items-center text-sm font-md bg-gray-100 text-gray-600 transition-colors hover:bg-gray-200 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-500 active:bg-gray-300 px-4 py-2 rounded-md false gap-2 h-[48px] font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
          >
            <svg
              aria-hidden="true"
              class="w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.636 5.636a9 9 0 1012.728 0M12 3v9"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="hidden md:block"
            >
              Sign Out
            </div>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-full flex-col px-3 py-4 md:px-2"
    >
      <a
        class="mb-2 flex h-20 items-end justify-start rounded-md bg-blue-600 p-4 md:h-40"
        href="/"
      >
        <div
          class="w-32 text-white md:w-40"
        >
          <div
            class="flex flex-row items-center leading-none text-white"
          >
            <p
              class="text-[24px]"
            >
              User Management
            </p>
          </div>
        </div>
      </a>
      <div
        class="flex grow flex-row justify-between space-x-2 md:flex-col md:space-x-0 md:space-y-2"
      >
        <a
          class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm text-black font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
          data-testid="/home"
          href="/home"
        >
          <svg
            aria-hidden="true"
            class="w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="hidden md:block"
          >
            Home
          </p>
        </a>
        <a
          class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm text-black font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
          data-testid="/users"
          href="/users"
        >
          <svg
            aria-hidden="true"
            class="w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 18.72a9.094 9.094 0 003.741-.479 3 3 0 00-4.682-2.72m.94 3.198l.001.031c0 .225-.012.447-.037.666A11.944 11.944 0 0112 21c-2.17 0-4.207-.576-5.963-1.584A6.062 6.062 0 016 18.719m12 0a5.971 5.971 0 00-.941-3.197m0 0A5.995 5.995 0 0012 12.75a5.995 5.995 0 00-5.058 2.772m0 0a3 3 0 00-4.681 2.72 8.986 8.986 0 003.74.477m.94-3.197a5.971 5.971 0 00-.94 3.197M15 6.75a3 3 0 11-6 0 3 3 0 016 0zm6 3a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0zm-13.5 0a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="hidden md:block"
          >
            Users
          </p>
        </a>
        <a
          class="flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm text-black font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
          data-testid="/techstacks"
          href="/techstacks"
        >
          <svg
            aria-hidden="true"
            class="w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.25 6.75L22.5 12l-5.25 5.25m-10.5 0L1.5 12l5.25-5.25m7.5-3l-4.5 16.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="hidden md:block"
          >
            Tech Stacks
          </p>
        </a>
        <div
          class="hidden h-auto w-full grow rounded-md bg-gray-50 md:block"
        />
        <button
          class="flex items-center text-sm font-md bg-gray-100 text-gray-600 transition-colors hover:bg-gray-200 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-500 active:bg-gray-300 px-4 py-2 rounded-md false gap-2 h-[48px] font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
        >
          <svg
            aria-hidden="true"
            class="w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.636 5.636a9 9 0 1012.728 0M12 3v9"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <div
            class="hidden md:block"
          >
            Sign Out
          </div>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
